drivers/frequency/adf5902/adf5902.c

```diff
@@ -0,0 +1,1046 @@
+ * Copyright 2021(c) Analog Devices, Inc.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <malloc.h>
+#include <math.h>
+#include "adf5902.h"
+#include "error.h"
+#include "delay.h"
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/**
+ * @brief Writes 4 bytes of data to ADF5902.
+ * @param dev - The device structure.
+ * @param data - Data value to write.
+ * @return Returns 0 in case of success or negative error code otherwise.
+ */
+int32_t adf5902_write(struct adf5902_dev *dev, uint32_t data)
```

Comment:
Is it possible to have register read ad write functions that have the standard signature?
Standard register read function: xxxxxx_reg_read(device *ptr, uint32_t addr, uint32_t *val)
Standard register write function: xxxxxx_reg_read(device *ptr, uint32_t addr, uint32_t val)

Reply:
The register addresses are 5-bit long, so it will require more data shifting and arrangement within the read/write functions to properly transmit/receive the bytes.

Moreover, all the other ADF* parts from this section are using this format:
https://github.com/analogdevicesinc/no-OS/blob/b7312f038fa915933b77142d86101dd024658a0c/drivers/frequency/adf4156/adf4156.h#L239
https://github.com/analogdevicesinc/no-OS/blob/b7312f038fa915933b77142d86101dd024658a0c/drivers/frequency/adf4157/adf4157.h#L233
https://github.com/analogdevicesinc/no-OS/blob/b7312f038fa915933b77142d86101dd024658a0c/drivers/frequency/adf4350/adf4350.h#L206

Reply:
Allways 4 bytes will be written? 
I agree with Andrei that the standard register function are a better approach.
Anyway I think it will require a lot of changes so I don't know what to say for this one.
But for the future I would use the standard ones